### PR TITLE
add structs to define input for functions

### LINKED ISSUE #1: Make data configurable
https://github.com/fatboypunk/lti_elixir/blob/f5bf86528af8a89d5606a3d520b691170f17faee/lib/lti_elixir.ex#L24-L43

### PR DESCRIPTION
**Pull request by fatboypunk**
this makes launch parameters variable, the structs enforces lti data.
fixes #1

### CODE REVIEW

File: lib/lti_elixir.ex [hunk 1]
```diff
@@ -2,72 +2,46 @@ defmodule LtiElixir do
...
-
-  def launch_params(params) do
-    launch_data = Enum.reduce(params, %{}, fn({key, value}, acc) ->
-      Map.put(acc, key,  value)
-    end)
-
-    Enum.reduce(launch_data, [], fn({key, value}, acc) ->
+  def launch_data(%OAuthData{} = oauth, %LaunchParams{} = launch_params) do
+    struct_to_list(oauth) ++ struct_to_list(launch_params)
+    |> Enum.reduce(%{}, fn({key, value}, acc) -> Map.put(acc, key,  value) end)
```
Comment (sn3p) [inline lib/lti_elixir.ex]:
```
key,  value
    ^^
```

File: lib/lti_elixir.ex [hunk 2]
```diff
@@ -85,4 +59,6 @@ defmodule LtiElixir do
     |> to_string()
     |> URI.encode(&URI.char_unreserved?/1)
   end
+
+  defp struct_to_list(struct), do: struct |> Map.from_struct() |> Map.to_list()
```
Comment (sn3p) [inline lib/lti_elixir.ex]: Pipe on new lines? Or no?

File: lib/lti_elixir.ex (hunk 1 above)
Comment (fatboypunk) [reply lib/lti_elixir.ex]: 😱

File: lib/lti_elixir.ex (hunk 2 above)
Comment (fatboypunk) [reply lib/lti_elixir.ex]: Sure

**Pull request merged**